projects/plugins/jetpack/_inc/client/portals/plugin-deactivation.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect, useCallback } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import PortalSidecar from './utilities/portal-sidecar';
+import {
+	getApiRootUrl,
+	getApiNonce,
+	getInitialStateConnectedPlugins,
+	getInitialStateJetpackBenefits,
+	getTracksUserData,
+} from '../state/initial-state';
+import restApi from '@automattic/jetpack-api';
+import { DisconnectDialog } from '@automattic/jetpack-connection';
+import JetpackBenefits from '../components/jetpack-benefits';
+
+/**
+ * Component that loads on the plugins page and manages presenting the disconnection modal.
+ *
+ * @param {object} props - The props object for the component.
+ * @param {string} props.apiRoot - Root URL for the API, which is required by the <DisconnectDialog/> component.
+ * @param {string} props.apiNonce - Nonce value for the API, which is required by the <DisconnectDialog/> component.
+ * @param {object} props.connectedPlugins - An object of plugins that are using the Jetpack connection.
+ * @param {Array} props.siteBenefits - An array of benefits provided by Jetpack.
+ * @param {string} props.pluginUrl - The URL of the plugin directory.
+ * @returns {React.Component} - The PluginDeactivation component.
+ */
+const PluginDeactivation = props => {
+	const { apiRoot, apiNonce, connectedPlugins, siteBenefits, tracksUserData } = props;
+	const [ modalOpen, setModalOpen ] = useState( false );
+
+	// Modify the deactivation link.
+	// deactivate_dialog is a localized variable.
+	const deactivate_dialog = window.deactivate_dialog;
+	const deactivationLink = document.querySelector( '#deactivate-jetpack, #deactivate-jetpack-dev' ); // ID set by WP on the deactivation link.
+
+	deactivationLink.setAttribute( 'title', deactivate_dialog.title );
+	deactivationLink.textContent = deactivate_dialog.deactivate_label;
+
+	useEffect( () => {
+		restApi.setApiRoot( apiRoot );
+		restApi.setApiNonce( apiNonce );
+	}, [ apiRoot, apiNonce ] );
+
+	const toggleVisibility = useCallback( () => {
+		setModalOpen( ! modalOpen );
+	}, [ setModalOpen, modalOpen ] );
+
+	const handleLinkClick = useCallback(
+		e => {
+			e.preventDefault();
+			toggleVisibility();
+		},
+		[ toggleVisibility ]
+	);
+
+	const handleLinkKeyDown = useCallback(
```

Comment:
Is handling key down events like this, in addition to click events, something that we normally do throughout the rest of the Jetpack plugin?

Reply:
Great question. I'm not entirely sure.
I was half convinced that I had added this key handler for a linter rule, but removing it does not trigger any errors. 
I think this can be removed.